sites/docs/docs/plugins/creating-a-plugin/datasources.md

```diff
@@ -0,0 +1,75 @@
+# Creating a Data Source Plugin
+
+The [Evidence postgres source plugin](https://github.com/evidence-dev/evidence/tree/main/packages/postgres) is a good
+reference for a source plugin.
+
+## Options Specification
+
+Evidence Datasources must provide an `options` export; this is used to
+build UI and validation to ensure an excellent UX for Evidence users.
+
+Options can have the following fields:
+
+| Field            | Type                                                    | Required | Description                                                                                                                                                                                         |
+|------------------|---------------------------------------------------------|----------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `title`          | string                                                  | ✅        | Name or Title of the option                                                                                                                                                                         |
+| `type`           | 'string' \| 'number' \| 'boolean' \| 'select' \| 'file' | ✅        | Control to show                                                                                                                                                                                     |
+| `secret`         | boolean                                                 | ❌        | Secret values are placed in `connection.options.yaml`, which is not source controlled                                                                                                               |
+| `shown`          | boolean                                                 | ❌        | Displays value in UI elements (e.g. for usernames, that should not be source controlled but are not "secret")                                                                                       |
```

Comment:
```suggestion
| `shown`          | boolean                                                 | ❌        | Displays value in UI elements (e.g. for usernames, that should not be source controlled but are not "secret". Otherwise the field will display as ∙∙∙)                                                                                       |
```